tutorials/3d/spring_arm.rst

```diff
@@ -0,0 +1,117 @@
+
+The "shape" of the spring arm is what it uses to check for collisions. The spring arm will "sweep" this shape from its origin out towards its length.
+
+.. image:: img/spring_arm_shape.svg
+
+The spring arm tries to keep all of its children at the end of its length. When the shape collides with something, the children are instead placed at or near that collision point:
+
+.. image:: img/spring_arm_children.svg
+
+Spring arm with a camera
+------------------------
+
+When a camera is placed as a child of a spring arm, a pyramid representing the camera will be used as the shape.
+
+This pyramid represents the **near plane** of the camera:
+
+.. image::: img/spring_arm_camera_shape.svg
+
+.. note:: If the spring arm is given a specific shape, then that shape will **always** be used.
+
+    The camera's shape is only used if the camera is a **direct child** of the spring arm.
+
+    If no shape is provided and the camera is not a direct child, the spring arm will fall back to using a ray cast which is inaccurate for camera collisions and not recommended.
+
+Every physics process frame, the spring arm will perform a motion cast to check if anything is collided with:
+
+.. image:: img/spring_arm_camera_motion_cast.svg
+
+When the shape hits something, the camera will be placed at the collision point:
+
+.. image:: img/spring_arm_camera_collision.svg
+
+Setting up the spring arm and camera
+------------------------------------
+
+Let's add a spring arm camera setup to the platformer demo.
+
+.. note:: You can download the Platformer3D demo on `GitHub <https://github.com/godotengine/godot-demo-projects/tree/master/3d/platformer>`_ or using the `Asset Library <https://godotengine.org/asset-library/asset/2748>`_.
+
+In general, for a third-person camera setup, you will have three nodes as children of the node that you're following:
+
+- `Node3D` (the "pivot point" for the camera)
+
+    - `SpringArm3D`
+
+        - `Camera3D`
+
+Set these up as children of our player and give them unique names so we can find them in our script:
+
+.. image:: img/spring_arm_editor_setup.png
+
+Let's move the pivot point up by `1.5` on the Y-axis so that it's not on the ground:
+
+.. image:: img/spring_arm_pivot_setup.png
+
+
+Give the spring arm a length of `3` so that it is placed behind the character:
+
+.. image:: img/spring_arm_length_setup.png
+
+.. note:: Leave the "Shape" of the spring arm as `<empty>`. This way, it will use the camera's pyramid shape.
+
+    If you want, you can also try other shapes - a sphere is a common choice since it slides smoothly along edges.
+
+Update the top of `player.gd` to grab the camera and the pivot points by their unique names:
+
+.. code-block:: gdscript
+
+    @onready var _camera := %Camera3D as Camera3D
+    @onready var _camera_pivot := %CameraPivot as Node3D
+
+Add an `_unhandled_input` function to check for camera movement and then rotate the pivot point accordingly:
+
+.. code-block:: gdscript
+
```

Comment:
```suggestion
.. code-block:: gdscript
    :caption: player.gd
```
Add filenames for code in captions (new since https://github.com/godotengine/godot-docs/pull/10151)